doc/fast-reboot/Fast-reboot_Flow_Improvements_HLD.md

```diff
@@ -70,10 +71,79 @@ When swss docker starts with the new kernel, all the port/LAG, vlan, interface,
 
 The restart of syncd docker should leave data plane intact until it starts again with the new kernel. After restart, syncd configures the HW with the state prior the reboot by all network applications.
 
+## 3.2 In-Service restart and upgrade from a vendor NOS to SONiC NOS
+
+Fast-reboot will finish successfully from a different NOS than SONiC with two possible scenarios:
+ - Dump files of default gateway, neighbors and fdb tables are provided to the new image in a format that meet the SONiC scheme, as SONiC does prior the reboot.
+   - On this scenario all should work exacly the same as the switch rebooted from SONiC to SONiC.
+
+ - Dump files of default gateway, neighbors and fdb tables are not provided to the new image as SONiC does prior the reboot.
+   - On this scenario fast-reboot will finish successfully, but with low performance since all neighbors and fdb entries will be created by the slow path.
```

Comment:
The claim here is that without dump files the performance will be degraded. What metrics are you referring to? 

Additionally, I see that without dump files, the downtime is >30s, and the test failed. I believe that is unexpected. 

`2022-05-18 15:56:33 : FAILED:dut:Total downtime period must be less then 0:00:30 seconds. It was 48.5044789314`

Reply:
What platform was tested here? Even for good-case scenario the downtime is 28s, which is concerningly close to 30s SLA.

Reply:
@vaibhavhd what do you mean by metrics?
This is the current results we have with the current implementation, these test results added to the HLD as you requested but actually not related to this new feature.
The new design will support the flow you are requesting but as for performance for the new flow, if this is not suffice it should be handled as a different feature, We can discuss it and add it as FR.

Reply:
By metrics I was meaning to ask what you meant by `low performance` in this line `On this scenario fast-reboot will finish successfully, but with low performance`

Reply:
Regarding the failure case (48s downtime), I did not notice that this was from current implementation. I assumed you tested it based on your local changes for new design.